docs: Update userguide (#2996)
probabl-ai/skore: sphinx/user_guide/index.rst

`skore` helps at structuring and storing what matters in your data science
experiments.

When it comes to data science, many libraries are available to help you investigate
your data. `pandas` or `polars` are great tools to explore and transform your data.

> `skrub` is the one tool that brings the necessary *statefullness* to those

Reviewer: ```suggestion
`skrub` is the one tool that brings the necessary *statefulness* to those
```
while we're at it

transformations required by a machine learning pipeline
(refer to `skrub's documentation <https://skrub-data.org/stable/documentation.html>`_).
`scikit-learn` and other `scikit-learn` compatible libraries (e.g. `xgboost`,
`lightgbm`) provide a set of algorithms to ingest transformed data and create
predictive models, as well as tools to diagnose and evaluate them.

All these libraries are broad and generic by design in order to accommodate a wide range 
of use cases. It is your experience in choosing the appropriate building blocks from
those libraries that is the key to success.

`skore` is the package that ties all these pieces together. It allows you to 
leverage your experience via a structured and robust framework
for understanding the impact of your analysis choices, with seamless integration
of all the above tools.

`skore` takes in the *full data science pipeline* created by assembling those
building blocks and provides **structured artifacts** that store the
information that matters for your particular use case. It minimizes unnecessary

> overhead spent mired in busy work, such as navigating through documentation, 

Reviewer: ```suggestion
overhead spent on busy work, such as navigating through documentation, 
```

Author: if you don't like "mired in", then I would suggest "on" as it sounds more natural in English

Reviewer: updated the suggestion
